api/v1beta1/hostedcluster_conditions.go

```diff
@@ -39,6 +39,10 @@ const (
 	// A failure here may require external user intervention to resolve. E.g. cloud provider perms were corrupted. E.g. the guest cluster was broken
 	// and kube resource deletion that affects cloud infra like service type load balancer can't succeed.
 	CloudResourcesDestroyed ConditionType = "CloudResourcesDestroyed"
+	// HostedClusterDestroyed indicates when a HostedCluster is being deleted but still needs to wait for a grace period.
```

Comment:
>// HostedClusterDestroyed indicates when a HostedCluster is being deleted

then this should be called HostedClusterDestroying which can be true/false, but then again that's already signalled by the deletionTimeStamp.

Reply:
The deletion timestamp indicates that a hosted cluster is destroying. This condition really means that it has finished destroying, but the resource is waiting for the grace period to disappear.